README.md

```diff
@@ -1,2 +1,239 @@
+other words, we accept that a project this large requires a piecemeal approach
+but we shall understand and avoid design dead ends that prevent either scaling
+or rigorous analysis of anonymity properties.
+
+We think mix networks occupy a sweet spot in which nodes run extremely efficiently but rigorous analysis remains possible, if challenging.  There are designs like Tor with more efficient designs, but they cannot provide rigorous anonymity properties.  There are also numerous academic schemes designed to support rigorous analysis, but at extreme sacrifices in efficiency.  We choose a Sphinx-like packet format because it’s efficient, adaptable enough, and has good security proofs.  
+
+We leave actually analyzing our scheme for future work in collaboration with academics.  We shall however use Poison mixing and cover traffic strategy similar to Loopix, which appears analyzable although academic work to do so remains ongoing.  We concede that mix networks impose latency on users, but any faster design would definitely not admit rigorous analysis of anonymity properties, and could not credibly claim to be stronger than Tor.  
+
+We shall tweak Sphinx slightly to accommodate both receiver anonymity and
+short-term message storage simultaneously, which may require updating its
+security proofs eventually. We consider short-term message storage essential to
+user experience and advise against doing it via a second layer protocol.
+
+We foresee the public key infrastructure (PKI) being the ultimate scaling
+bottleneck for all existing anonymity scheme designs. We could avoid this with
+gossip protocols but these enable epistemic attacks. We largely leave this to
+future work, but suggest investigating a verifiable gossip based protocol. We
+shall therefore use an insecure gossip based protocol initially with the hope
+that it meshes best with later designs. If we used a more secure design inspired
+by the Tor consensus then we might assumptions elsewhere that limit scalability
+later.
+
+We know rewards for node operators remains a contentious question in the
+anonymity community with seemingly unforeseeable consequences. We nevertheless
+think rewards represents our best hope for a network large enough to challenge
+today’s centralized providers that operate on surveillance capitalism. We do not
+imagine rewards obliviate the need to steward relay operator culture, possibly
+quite the opposite.
+
+### Messaging types
+We’re focusing on one-to-one messaging for now.  We actually do require messaging layer crypto, even after all the mix net layers, so expect an Axolotl-like ratchet for this.  
+
+We can adapt our short-term message storage plans for small group messaging, but
+not with exactly the same privacy assurances. We leave designing this to future
+work.
+
+We think one-to-mass messaging should be done by using the mix network to send
+to a broadcast protocol like Whisper v1 or perhaps a blockchain.
+
+### Payment
```

Comment:
Not clear to me what extent payment should be requirement at this layer, vs something that's on top. How this looks like isn't clear to me. Some form of QoS optional differentiator, similar to Bittorrent choking/unchoking. But reputation should be over longer than just a torrent file - what this implies for design seems open to me.

From Status point of view, we'd love to see staking with our native SNT token for Status nodes (desktop, mobile and some headless nodes), but this would obviously not be part of a base layer. The more we can push this to above layers, or at least keep specifics of how this is done, the better, as that would allow individual projects to choose to do incentivization or not. It'd also be open to non blockchain projects, and more likely to survive as an open protocol of this isn't required.

Reply:
A naive QoS scheme breaks anonymity, but you could have QoS mean larger packet size, so you actually pay to have less anonymity but do things faster. 

There are several proposed payment schemes:
 - Validity Labs wants payment per packet with payment channels.  This requires altering the packet format dramatically.  I failed to break their most recent proposal, but I've not looked into enough yet really..  And DoS vulnerabilities sound likely.
 - Panoramix has nodes run by a cartel of service providers, who control access like email providers do.  This is not distributed in any sense we like.  Panoramix might eventually address this by switching to rerandomizable credentials ala Coconut, which requires altering the packet format. 
 - We favor "printing money" based on turning some cover traffic in "secret shopper" packets, because this help address the token distribution problem in a PoS system.  It'll seemingly run entirely separately.

We can initially build with no payment layer first, but plan for adding cover traffic and packet sizes.  We add the "secret shopper" scheme only after we add cover traffic.  We can introduce larger packet sizes with rerandomizable credentials later.